ThumbnailSequence.cpp

```diff
@@ -1560,7 +1560,7 @@ ThumbnailSequence::CompositeItem::CompositeItem(
 bool
 ThumbnailSequence::CompositeItem::incompleteThumbnail() const
 {
-	return dynamic_cast<IncompleteThumbnail*>(m_pThumb) != 0;
+	return m_pThumb == 0 && dynamic_cast<IncompleteThumbnail*>(m_pThumb) != 0;
```

Comment:
Looks like something is broken with FreeBSD. Maybe a better idea would be creating a small example reproducing the problem and let FreeBSD developers deal with it.

As for double-casting to IncompleteThumbnail, I would prefer a cleaner solution:
Make ThumbnailFactory return ThumbnailBase rather than QGraphicsItem and add a virtual isIncomplete() method to ThumbnailBase and IncompleteThumbnail.